drivers/adc/ad463x/ad463x.c

```diff
@@ -333,15 +343,35 @@ static int32_t ad463x_init_gpio(struct ad463x_dev *dev,
 		no_os_mdelay(100);
 	}
 
+	/** Configure PGIA pins, if available */
+	if (dev->has_pgia) {
+		ret = no_os_gpio_get_optional(&dev->gpio_pgia_a0,init_param->gpio_pgia_a0);
+		if (ret != 0)
+			return ret;
+		ret = no_os_gpio_get_optional(&dev->gpio_pgia_a1, init_param->gpio_pgia_a1);
+		if (ret != 0)
+			return ret;
```

Comment:
no_os_gpio_get_optional allocates memory dynamically. in case of an error it  should be freed before returning, otherwise there will be memory leaks